src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMProbe.java

```diff
@@ -151,20 +153,39 @@ public SCMProbeStat stat(@NonNull String path) throws IOException {
                     return SCMProbeStat.fromAlternativePath(content.getPath());
                 }
             }
-        } catch (FileNotFoundException fnf) {
+        } catch (GHFileNotFoundException fnf) {
+            boolean finicky = false;
             if (index == 0 || index == 1) {
-                // the revision does not exist, we should complain.
-                throw fnf;
+                // the revision does not exist, we should complain unless JENKINS-54126.
+                finicky = true;
             } else {
                 try {
                     repo.getDirectoryContent("/", Constants.R_REFS + ref);
                 } catch (IOException e) {
-                    // this must be an issue with the revision, so complain
+                    // this must be an issue with the revision, so complain unless JENKINS-54126
                     fnf.addSuppressed(e);
-                    throw fnf;
+                    finicky = true;
                 }
                 // means that does not exist and this is handled below this try/catch block.
             }
+            if (finicky) {
+                final Optional<List<String>> status = Optional.ofNullable(fnf.getResponseHeaderFields() != null ? fnf.getResponseHeaderFields().get(null) : null);
```

Comment:
```suggestion
                final Optional<List<String>> status = Optional.ofNullable(fnf.getResponseHeaderFields() != null ? Optional.ofNullable(fnf.getResponseHeaderFields().get(null)) : Optional.empty());
```
Turning @stephenc 's comment into a suggestion.

Reply:
That didn't help. It was because it thought that just because the method didn't return null on the first call it could on the second call :man_facepalming: .